activerecord/test/cases/unsafe_raw_sql_test.rb

```diff
@@ -114,6 +114,39 @@ class UnsafeRawSqlTest < ActiveRecord::TestCase
     end
   end if current_adapter?(:PostgreSQLAdapter)
 
+  def setup
+    @connection = ActiveRecord::Base.connection
+    @connection.create_table("json_data_type") do |t|
+      t.public_send :json, "payload", default: {} # t.json 'payload', default: {}
+      t.public_send :json, "settings"             # t.json 'settings'
+      t.public_send :json, "objects", array: true # t.json 'objects', array: true
+    end
+  end if current_adapter?(:PostgreSQLAdapter)
+
+  class JsonDataType < ActiveRecord::Base
+    self.table_name = "json_data_type"
+
+    store_accessor :settings, :resolution
+  end
+
+  test "order: allow JSON operators" do
+    created = JsonDataType.create!(payload: { "a" => 1, "b" => 2 })
+    jdt = JsonDataType.order("payload ->> 'a'").first
+    assert_equal created.id, jdt.id
+  end if current_adapter?(:PostgreSQLAdapter)
+
+  test "order: allow JSON operators with weird spacing" do
+    created = JsonDataType.create!(payload: { "a" => 1, "b" => 2 })
+    jdt = JsonDataType.order("payload::json->> 'a'").first
+    assert_equal created.id, jdt.id
+  end if current_adapter?(:PostgreSQLAdapter)
+
+  test "order: allow JSON operators with array" do
+    created = JsonDataType.create!(payload: ["hello", "world"])
+    jdt = JsonDataType.order("payload ->>1").first
+    assert_equal created.id, jdt.id
+  end if current_adapter?(:PostgreSQLAdapter)
```

Comment:
I think we should wrap all these tests in the block below, rather than having it after every test, wdyt?

```
if current_adapter?(:PostgreSQLAdapter)
  # all tests here
end
```

Reply:
In fact we could prob create a new class at the end of this file, maybe better worded 🙂:

```
if current_adapter?(:PostgreSQLAdapter)
class JsonOrdersTest < ActiveRecord::TestCase
  class JsonDataType < ActiveRecord::Base
    ...
  end
end

def setup...
```

Or maybe there is another way to simplify that but having the setup method for those 3 tests seems out of place?

Reply:
Done!

Reply:
Oh, sorry, the if statement is done. I hadn't refreshed.